App.js

```diff
@@ -0,0 +1,93 @@
+import React from "react"
+import Sidebar from "./components/Sidebar"
+import Editor from "./components/Editor"
+import { data } from "./data"
+import Split from "react-split"
+import {nanoid} from "nanoid"
+
+export default function App() {
+    const [notes, setNotes] = React.useState(JSON.parse(localStorage.getItem('Notes')) || [])
+    const [currentNoteId, setCurrentNoteId] = React.useState(
+        (notes[0] && notes[0].id) || ""
+    )
+    localStorage.setItem('Notes', JSON.stringify(notes))
+
+       
+    function deleteNote(event, noteId) {
+        event.stopPropagation()
+         setNotes(notes=>notes.filter(el=>el.id!==noteId))
+    }
+
+    function createNewNote() {
+        const newNote = {
+            id: nanoid(),
+            body: "# Type your markdown note's title here"
+        }
+        setNotes(prevNotes => [newNote, ...prevNotes])
+        setCurrentNoteId(newNote.id)
+    }
+    
+    function updateNote(text) {
+
+        setNotes(oldNotes => {
+
+          const newNotes = oldNotes.map(oldNote => {
+            return oldNote.id === currentNoteId
+                ? {id:oldNote.id, body: text }
```

Comment:
As you only update the text, you don't need to repeat the id here, you can just spread the old note and update the text.